tests/TestCase/Database/QueryTest.php

```diff
@@ -4376,12 +4376,8 @@ public function testSqlCaseStatement()
 
         //Postgres requires the case statement to be cast to a integer
         if ($this->connection->getDriver() instanceof \Cake\Database\Driver\Postgres) {
-            $publishedCase = $query->func()
-                ->cast([$publishedCase, 'integer' => 'literal'])
```

Comment:
Same feedback as the 4.next pull about backwards compatibility issue.

Reply:
I added backward compatibility with deprecation warning message.